apps/speed.c

```diff
@@ -561,52 +639,514 @@ static OPT_PAIR ecdh_choices[] = {
 };
 #endif
 
+#ifndef SIGALRM
+# define COND(d) (count < (d))
+# define COUNT(d) (d)
+#else
+# define COND(c) (run && count<0x7fffffff)
+# define COUNT(d) (count)
+#endif                         /* SIGALRM */
+
+static unsigned char *buf = NULL, *buf2 = NULL;
+static int j;
+static char *engine_id = NULL;
+
+
+#ifndef OPENSSL_NO_MD2
+static unsigned char md2[MD2_DIGEST_LENGTH];
```

Comment:
I think there is a risk here of things going wrong because the same variable is being shared across multiple jobs. We probably need to look at each variable in turn and consider what the impact is of it being shared. In this particular case I don't think your argument holds. This variable is *only* used by the loop function and never initialised anywhere else.